torch/csrc/jit/serialization/pickler.cpp

```diff
@@ -353,6 +353,41 @@ void Pickler::pushTensor(const IValue& ivalue) {
   }
 }
 
+void Pickler::pushLiteralSparseTensor(const at::Tensor& tensor) {
+  TORCH_INTERNAL_ASSERT(tensor.is_sparse());
+  // The arguments to this function are:
+  // size, requires_grad, pinned memory, data_type, device,
+  // indices, values, backward_hooks
+  pushGlobal("torch._utils", "_rebuild_sparse_coo_tensor");
+  // size
+  push<PickleOpCode>(PickleOpCode::MARK);
+  push<PickleOpCode>(PickleOpCode::MARK);
+  for (auto size : tensor.sizes()) {
+    pushInt(size);
+  }
+  push<PickleOpCode>(PickleOpCode::TUPLE);
+  // requires_grad
+  pushIValue(tensor.requires_grad());
+  // pinned memory
+  pushBool(tensor.options().pinned_memory());
+  // data type
+  pushInt(static_cast<uint16_t>(tensor.scalar_type()));
```

Comment:
what about memory format? should we also save that or sparse_coo is the only format that sparse tensor gonna support?

Reply:
The other sparse tensor format CSR. I did not update the serialization to support CSR. Right now the only sparse tensor format that will work is COO. So I did not save the memory format.